rasa/core/policies/rule_policy.py

```diff
@@ -212,15 +242,137 @@ def _create_loop_unhappy_lookup_from_states(
             ):
                 lookup[feature_key] = DO_NOT_VALIDATE_LOOP
             elif (
-                # some action other than action_listen and active_loop
-                # is predicted in unhappy path,
+                # some action other than active_loop is predicted in unhappy path,
                 # therefore active_loop shouldn't be predicted by the rule
                 not is_prev_action_listen_in_state(states[-1])
-                and action not in {ACTION_LISTEN_NAME, active_loop}
+                and action != active_loop
             ):
                 lookup[feature_key] = DO_NOT_PREDICT_LOOP_ACTION
         return lookup
 
+    def _check_rule_restriction(
+        self, rule_trackers: List[TrackerWithCachedStates]
+    ) -> None:
+        rules_exceeding_max_user_turns = []
+        for tracker in rule_trackers:
+            number_of_user_uttered = sum(
+                isinstance(event, UserUttered) for event in tracker.events
+            )
+            if number_of_user_uttered > self.ALLOWED_NUMBER_OF_USER_INPUTS:
+                rules_exceeding_max_user_turns.append(tracker.sender_id)
+
+        if rules_exceeding_max_user_turns:
+            raise InvalidRule(
+                f"Found rules '{', '.join(rules_exceeding_max_user_turns)}' "
+                f"that contain more than {self.ALLOWED_NUMBER_OF_USER_INPUTS} "
+                f"user inputs. Rules are not meant to hardcode a state machine. "
+                f"Please use stories for these cases."
```

Comment:
```
rasa.core.policies.rule_policy.InvalidRule: Found rules 'Greet' that contain more than 1 user inputs. Rules are not meant to hardcode a state machine. Please use stories for these cases.
You can find more information about the usage of rules at https://rasa.com/docs/rasa/core/rules/. 
```
the error message is on the same line as `rasa.core.policies.rule_policy.InvalidRule` - can we make it so that it comes on a new line afterwards?

Reply:
also would remove the sentence "Rules are not meant to hardcode a state machine"

Reply:
also not sure where the url is coming from, but the correct url will actually be: https://rasa.com/docs/rasa/rules